### DISCUSSION TITLE
Best practice for AuthN and AuthZ

### DISCUSSION THREAD

**Discussion by margaretjoanmiller** [99designs/gqlgen#3320] [Q&A]
I was wondering how folks are handling authentication and authorization and how each approach went. I need to restrict certain data to specific roles.

**Comment by fasibio** [comment]
Simple use an Auth graphql directive? 
There you can also check for roles. 
But this is not gqlgen specific is graphql standard way

**Reply by margaretjoanmiller** [reply]
Like this but implemented in go? https://github.com/graphql-community/graphql-directive-auth? I can't see from the official specs an official auth directive

**Comment by fasibio** [comment]
Mhm more or less. But in general you have the Auth token as header by your request. 
Than on server side you have to read them and interpret them. 
Did you read this https://gqlgen.com/reference/directives/ ? How directives work in gerneral?
